src/V3Number.cpp

```diff
@@ -941,6 +950,50 @@ int V3Number::widthMin() const {
     return 1;  // one bit even if number is == 0
 }
 
+uint32_t V3Number::countBits(const V3Number& ctrl) const {
+    int n = 0;
+
+    switch (ctrl.bitIs(0)) {
+        case '0': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIs0(bit)) n++;
+            }
+            break;
+        }
+
+        case '1': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIs1(bit)) n++;
+            }
+            break;
+        }
+
+        case 'x': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIsX(bit)) n++;
+            }
+            break;
+        }
+
+        case 'z': {
+        for (int bit = 0; bit < this->width(); bit++) {
+                if (bitIsZ(bit)) n++;
+            }
+            break;
+        }
+    }
+    return n;
+}
```

Comment:
Suggest moving loop so less code, and prefer prefix increment.

```suggestion
uint32_t V3Number::countBits(const V3Number& ctrl) const {
    int n = 0;
    for (int bit = 0; bit < this->width(); ++bit) {
        switch (ctrl.bitIs(0)) {
        case '0':
            if (bitIs0(bit)) ++n;
            break;
        case '1':
            if (bitIs1(bit)) ++n;
            break;
        case 'x':
            if (bitIsX(bit)) ++n;
            break;
        case 'z':
            if (bitIsZ(bit)) ++n;
            break;
        }
    }
    return n;
}
```